api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -422,7 +422,8 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
               value -> {
                 if (!value.equals(EMPTY_CELL_MARKER)) {
                   Double fieldValue = Double.parseDouble(value);
-                  queryValues.add(dateFormat.format(new Date(fieldValue.longValue())));
+                  // Bigquery returns time values in seconds, rather than in milliseconds.
+                  queryValues.add(dateFormat.format(new Date(fieldValue.longValue() * 1000)));
```

Comment:
You shouldn't have to do conversions inline here. There are things like [ofEpochSecond](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#ofEpochSecond-long-int-java.time.ZoneOffset-)

Reply:
or use [Instant](https://www.tutorialspoint.com/javatime/javatime_instant_ofepochsecond.htm)